stdlib/public/core/Collection.swift

```diff
@@ -24,7 +24,8 @@ public protocol Indexable {
   ///
   /// Valid indices consist of the position of every element and a
   /// "past the end" position that's not valid for use as a subscript.
-  associatedtype Index : ForwardIndex
+// TODO: swift-3-indexing-model - Index only needs to be comparable or must be comparable..?
```

Comment:
It was a question in my head along those lines. I can't think of one either at the moment but it is in the back of my mind. Basically current ForwardIndex only requires Equatable.